histogramvec/histogramvec.go

```diff
@@ -87,7 +87,16 @@ func (hv *HistogramVec) Ensure(labels []string) {
 	}
 }
 
-// Histograms returns the currently managed Histograms.
+// Histograms returns a copy of the currently managed Histograms.
 func (hv *HistogramVec) Histograms() map[string]*histogram.Histogram {
-	return hv.histograms
+	hv.mutex.Lock()
+	defer hv.mutex.Unlock()
+
+	histogramCopy := map[string]*histogram.Histogram{}
+
+	for label, histogram := range hv.histograms {
+		histogramCopy[label] = histogram
```

Comment:
Here you only get a different map holding the same pointers if I'm not mistaken. If you want to make a full copy here you have to get a new pointer for each `histogram`.

```go
h:=*histogram
histogramCopy[label] = &h
```

Reply:
That does right to me.

Would the new pointer just point to the original struct though? I feel like I need to copy the entire struct.

Reply:
The new pointer (`&h`) would point to the copied value (`h`) and no longer the original one (`*histogram`). So no need for further copy.